+nix/SourcesMixIn.m

```diff
@@ -13,27 +13,37 @@
     properties (Abstract, Hidden)
         alias
     end
-    
+
     methods
         function obj = SourcesMixIn()
             nix.Dynamic.add_dyn_relation(obj, 'sources', @nix.Source);
         end
         
+        % has_source supports only check by id, not by name
+        function ret = has_source(obj, id_or_entity)
+            has = id_or_entity;
+            if(strcmp(class(has), 'nix.Source'))
+            	has = id_or_entity.id;
+            end;
+            ret = nix_mx(strcat(obj.alias, '::hasSource'), ...
+                            obj.nix_handle, has);
+        end
```

Comment:
Since we are using id basically everywhere in the mx project, i'd change everything to name where it makes sense in one go.